src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsTextHelper.cs

```diff
@@ -101,7 +101,21 @@ private static string HumanizeCodeTags(this string text)
 
         private static string HumanizeMultilineCodeTags(this string text)
         {
-            return MultilineCodeTag().Replace(text, (match) => "```" + match.Groups["display"].Value + "```");
+            return MultilineCodeTag().Replace(text, match =>
+            {
+                var codeText = match.Groups["display"].Value;
+                if (LineBreaks().IsMatch(codeText))
+                {
+                    if (codeText.StartsWith("\r") || codeText.StartsWith("\n"))
+                    {
+                        return $"```{codeText.TrimEnd()}{Environment.NewLine}```";
+                    }
+
+                    return $"```{Environment.NewLine}{codeText.TrimEnd()}{Environment.NewLine}```";
```

Comment:
What about something like this:

```suggestion
                    var builder = new StringBuilder()
                        .Append("```");

                    if (!codeText.StartsWith("\r") && !codeText.StartsWith("\n"))
                    {
                        builder.AppendLine();
                    }

                    builder.AppendLine(codeText.TrimEnd())
                           .Append("```");

                    return builder.ToString();
```

Reply:
applied suggestion